pandas/core/index.py

```diff
@@ -1548,8 +1548,15 @@ def sym_diff(self, other, result_name=None):
         other, result_name_update = self._convert_can_do_setop(other)
         if result_name is None:
             result_name = result_name_update
-        the_diff = sorted(set((self.difference(other)).union(other.difference(self))))
-        return Index(the_diff, name=result_name)
+        the_diff = self.difference(other).union(other.difference(self))
+        the_diff_sorted = np.sort(the_diff.unique())
```

Comment:
do the same as above. If something specific is breaking (e.g. a specific type of index), then post it. You should NOT use the default numpy functions except where you know exactly what you have. They tend to handle things in very odd ways actually.